src/lib.rs

```diff
@@ -37,14 +37,191 @@ impl Parse for InputStruct {
     }
 }
 
+struct BrainfunctProgram {
+    operations: Vec<u8>,
+    functions: Vec<usize>,
+    main_index: usize,
+    input: Vec<u8>,
+    output: Vec<u8>,
+}
+
+impl Parse for BrainfunctProgram {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let p: LitStr = input.parse()?;
+        let p_string = p.value();
+        input.parse::<Token![,]>()?;
+        let input_str: LitStr = input.parse()?;
+        let input_string = input_str.value();
+        input.parse::<Token![,]>()?;
+        let output: LitStr = input.parse()?;
+        let output_string = output.value();
+        let mut temp = BrainfunctProgram {
+            operations: Vec::with_capacity(p_string.len()),
+            functions: vec![usize::MAX],
+            main_index: usize::MAX,
+            input: Vec::with_capacity(input_string.len()),
+            output: Vec::with_capacity(output_string.len()),
+        };
+        let mut func_index = 0;
+        for (index, c) in p.value().bytes().enumerate() {
+            temp.operations.push(match c {
+                b'>' | b'<' | b'+' | b'-' | b'.' | b',' | b'@' => c,
+                b'/' => {
+                    temp.functions.push(temp.main_index);
+                    temp.main_index = index;
+                    func_index += 1;
+                    c
+                }
+                _ => {
+                    return Err(syn::Error::new(
+                        Span::call_site(),
+                        "You had an invalid character in your program",
+                    ))
+                }
+            });
+
+            // The value is 256 because you can have 255 functions (and they start indexing at 1)
+            // and a main function which will cause the len to be 256 in the greatest case
+            if temp.functions.len() > 256 {
+                return Err(syn::Error::new(
+                    Span::call_site(),
+                    "You had more functions than are able to be called",
+                ));
+            }
+        }
+
+        if func_index == 0 {
+            return Err(syn::Error::new(
+                Span::call_site(),
+                "You didn't define any functions at all",
+            ));
+        }
+
+        for c in input_string.bytes() {
+            temp.input.push(c);
+        }
+
+        for c in output_string.bytes() {
+            temp.output.push(c);
+        }
+
+        Result::Ok(temp)
+    }
+}
+
+struct BrainfunctArray {
+    value: [u8; u16::MAX as usize],
```

Comment:
again, heh?

Reply:
What are you hehing at?

Reply:
is that instantiating an array of max usize or an array filled wjth max values?

Reply:
It is defining the value as being an array of size u16::MAX full of u8's

Reply:
ok so yep that's why i was slightly confused 😅 beeg array